### PR TITLE
Added the ability to override the replication logger

### PR DESCRIPTION
**Pull request by snowzach**
The replication module has logrus hard-coded. In some cases we may want it to integrate with something else. This PR adds the ability to pass something that resembles logrus that you can pipe logs through. If you do not specify it as part of the syncer config, it defaults to using logrus just as before.

### CODE REVIEW

**Comment by snowzach** [comment]
@siddontang I updated all of the logrus references in the project to point to  https://github.com/birkirb/loggers instead which allows easy overriding of the logger library. I'm not real good with dep yet so I couldn't quite figure out how to clean that stuff up. I think it's okay in it's current form but you may want to check it.

File: Gopkg.toml [hunk 1]
```diff
@@ -50,7 +50,7 @@
   name = "github.com/siddontang/go"
 
 [[constraint]]
-  name = "github.com/sirupsen/logrus"
+  name = "gopkg.in/birkirb/loggers.v1"
   version = "1.0.3"
```
Comment (siddontang) [inline Gopkg.toml]: the same version for loggers?

**Comment by siddontang** [comment]
@snowzach 

You can use `dep ensure -add gopkg.in/birkirb/loggers.v1` to add the vendor and then you can call `./clear_vendor.sh`

**Comment by siddontang** [comment]
I also guess you can remove the logrus now?

**Comment by siddontang** [comment]
Hi @snowzach 

Seem it is fine, can you add prune in the toml file https://golang.github.io/dep/docs/Gopkg.toml.html#prune

```
[prune]
  go-tests = true
  unused-packages = true
  none-go = true
```

**Comment by snowzach** [comment]
@siddontang sure. It was there already, I added non-go

**Pull request merged**